pkg/utils/utils.go

```diff
@@ -423,3 +431,69 @@ func GetPfEswitchMode(pciAddr string) (string, error) {
 	}
 	return devLinkDeviceAttrs.Mode, nil
 }
+
+// HasDefaultRoute returns true if PCI network device is default route interface
+func HasDefaultRoute(pciAddr string) (bool, error) {
+	// Get net interface name
+	ifNames, err := GetNetNames(pciAddr)
+	if err != nil {
+		return false, fmt.Errorf("error trying get net device name for device %s", pciAddr)
+	}
+
+	if len(ifNames) > 0 { // there's at least one interface name found
+		for _, ifName := range ifNames {
+			routes, err := GetNetlinkProvider().GetIPv4RouteList(ifName) // IPv6 routes: all interface has at least one link local route entry
+			if err != nil {
+				glog.Errorf("failed to get routes for interface: %s, %q", ifName, err)
+				continue
+			}
+			for _, r := range routes {
+				if r.Dst == nil {
+					glog.Infof("excluding interface %s:  default route found: %+v", ifName, r)
+					return true, nil
+				}
+			}
+		}
+	}
+
+	return false, nil
+}
+
+// NormalizeVendorName returns vendor name cropped to fit into the length of maxVendorName if it is bigger
+func NormalizeVendorName(vendor string) string {
+	vendorName := vendor
+	if len(vendor) > maxVendorName {
+		vendorName = string([]byte(vendorName)[0:17]) + "..."
+	}
+	return vendorName
+}
+
+// NormalizeProductName returns product name cropped to fit into the length of maxProductName if it is bigger
+func NormalizeProductName(product string) string {
+	productName := product
+	if len(product) > maxProductName {
+		productName = string([]byte(productName)[0:37]) + "..."
+	}
+	return productName
+}
+
+// ParseDeviceID returns device ID parsed from the string as 64bit integer
+func ParseDeviceID(deviceID string) (int64, error) {
+	return strconv.ParseInt(deviceID, classIDBaseInt, classIDBitSize)
+}
+
+// GetAuxNetDevicesFromPci returns list of auxiliary devices names for the specified PCI address
+func GetAuxNetDevicesFromPci(pciAddr string) ([]string, error) {
```

Comment:
so, are we going to use : https://github.com/k8snetworkplumbingwg/sriovnet/pull/55 ?

if yes, this should be deleted.

Reply:
Good catch. I probably forgot about this file.